src/linear_solver_logs/solve_log_full_ma.jl

```diff
@@ -0,0 +1,248 @@
+    sampler::Union{DataType,Nothing}
+    max_dimension::Int64
+    sigma2::Union{Float64, Nothing}
+    omega::Union{Float64, Nothing}
+    eta::Float64
+    true_res::Bool
+end
+
+LSLogFullMA() = LSLogFullMA(
+                          1,
+                          MAInfo(1, 30, 1, false, 1, zeros(30)),
+                          Float64[], 
+                          Float64[], 
+                          Int64[],
+                          norm, 
+                          -1, 
+                          false,
+                          nothing,
+                          0,
+                          nothing,
+                          nothing,
+                          1,
+                          false
+                         )
+
+LSLogFullMA(;lambda1 = 1, lambda2 = 30, sigma2 = nothing, omega = nothing, eta = 1, true_res = false) = LSLogFullMA(
+                          1,
+                          MAInfo(1, lambda2, 1, false, 1, zeros(lambda2)),
+                          Float64[], 
+                          Float64[], 
+                          Int64[],
+                          norm, 
+                          -1, 
+                          false,
+                          nothing,
+                          0,
+                          sigma2,
+                          omega,
+                          eta, 
+                          true_res
+                         )
+
+#Function to update the moving average
+function log_update!(
+    log::LSLogFullMA,
+    sampler::LinSysSampler,
+    x::AbstractVector,
+    samp::Tuple,
+    iter::Int64,
+    A::AbstractArray,
+    b::AbstractVector
+)
+    log.max_dimension = maximum(size(A))
+    ma_info = log.ma_info
+    log.iterations = iter
+    log.sampler = typeof(sampler)  
+    if iter != 0
+        #Check if we want exact residuals computed
+        if !log.true_res
+            # Compute the current residual to second power to align with theory
+            res::Float64 = log.resid_norm(samp[3])^2 
+        elseif typeof(sampler) <: LinSysVecColSampler 
+            res = log.resid_norm(A'* (A * x - b))^2
+        else
+            res = log.resid_norm(A * x - b)^2 
+        end
+        # Check if MA is in lambda1 or lambda2 regime
+        if ma_info.flag
+            Update_MAEstimators!(log, ma_info, res, ma_info.lambda2, iter)
+        else
+            #Check if we can switch between lambda1 and lambda2 regime
+            if res < ma_info.res_window[ma_info.idx]
+                Update_MAEstimators!(log, ma_info, res, ma_info.lambda1, iter)
+            else
+                Update_MAEstimators!(log, ma_info, res, ma_info.lambda1, iter)
+                ma_info.flag = true 
+            end
+
+        end
+
+    end
+
+end
+
+# Update the moving average estimator requires the log variable, ma_info,
+# observed residual, and a lambda_base which corresponds to which lambda regime we are in. 
+function Update_MAEstimators!(log, ma_info, res, lambda_base, iter)
```

Comment:
Does this follow the BLUE style convention?

Reply:
Also, why are we not passing types here?

Reply:
Docstring?

Reply:
I believe we have been following a convention that if a function is not exported no docstrings are added because the user should not be using non-exported functions. I added the types and updated the name to align with BLUE